### PR TITLE
Add Nucleo-F439ZI BSP

### PR DESCRIPTION
**Pull request by utzig**

### CODE REVIEW

File: hw/bsp/nucleo-f439zi/src/hal_bsp.c [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+#include <uart/uart.h>
+#include <uart_hal/uart_hal.h>
+#endif
+
+#include <hal/hal_bsp.h>
+#include <hal/hal_gpio.h>
+#include <hal/hal_timer.h>
+
+#include <stm32f4xx_hal_gpio_ex.h>
+#include "mcu/stm32_hal.h"
```
Comment (kasjer) [inline hw/bsp/nucleo-f439zi/src/hal_bsp.c]: out of curiosity: why "" and <> ?

File: hw/bsp/nucleo-f439zi/include/bsp/bsp.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#define ARDUINO_PIN_D8      MCU_GPIO_PORTF(12)
+#define ARDUINO_PIN_D9      MCU_GPIO_PORTD(15)
+#define ARDUINO_PIN_D10     MCU_GPIO_PORTD(14)
+#define ARDUINO_PIN_D11     MCU_GPIO_PORTA(7)
+#define ARDUINO_PIN_D12     MCU_GPIO_PORTA(6)
+#define ARDUINO_PIN_D13     MCU_GPIO_PORTA(5)
+#define ARDUINO_PIN_A0      MCU_GPIO_PORTA(3)
+#define ARDUINO_PIN_A1      MCU_GPIO_PORTC(0)
+#define ARDUINO_PIN_A2      MCU_GPIO_PORTC(3)
+#define ARDUINO_PIN_A3      MCU_GPIO_PORTC(1)
```
Comment (kasjer) [inline hw/bsp/nucleo-f439zi/include/bsp/bsp.h]: are you sure A3-A6 are correct?

File: hw/bsp/nucleo-f439zi/src/hal_bsp.c (hunk 1 above)
Comment (utzig) [reply hw/bsp/nucleo-f439zi/src/hal_bsp.c]: It's the result of a powerful feature called "copy and paste"; should be using <> now.

File: hw/bsp/nucleo-f439zi/include/bsp/bsp.h (hunk 2 above)
Comment (utzig) [reply hw/bsp/nucleo-f439zi/include/bsp/bsp.h]: I surely am not, it was incorrect and it's fixed now, thanks!

**Pull request merged**